setup.py

```diff
@@ -1,42 +1,151 @@
+import distutils
 import glob
 import os
+import shlex
 import setuptools
 import setuptools.command.build_ext
+import setuptools.command.develop
+import setuptools.command.install
 import setuptools.dist
 import sys
 import setuptools.command.install
 import cellprofiler.utilities.version
 
+# Recipe needed to get real distutils if virtualenv.
+# Error message is "ImportError: cannot import name dist"
+# when running app.
+# See http://sourceforge.net/p/py2exe/mailman/attachment/47C45804.9030206@free.fr/1/
+#
+if hasattr(sys, 'real_prefix'):
+    # Running from a virtualenv
+    assert hasattr(distutils, 'distutils_path'), \
+        "Can't get real distutils path"
+    libdir = os.path.dirname(distutils.distutils_path)
+    sys.path.insert(0, libdir)
+    #
+    # Get the system "site" package, not the virtualenv one. This prevents
+    # site.virtual_install_main_packages from being called, resulting in
+    # "IOError: [Errno 2] No such file or directory: 'orig-prefix.txt'
+    #
+    del sys.modules["site"]
+    import site
+
+    assert not hasattr(site, "virtual_install_main_packages")
+
 setuptools.dist.Distribution({
     "setup_requires": [
         "clint",
+        "javabridge",
+        "matplotlib",
+        "numpy",
+        "pytest",
         "requests",
-        "pytest"
+        "scipy",
+        "pyzmq"
     ]
 })
 
+try:
+    import matplotlib
+    import numpy  # for proper discovery of its libraries by distutils
+    import scipy.sparse.csgraph._validation
+    import scipy.linalg
+    import zmq  # for proper discovery of its libraries by distutils
+    import zmq.libzmq
+except ImportError:
+    pass
+
+if sys.platform.startswith("win"):
+    import _winreg
+
+    try:
+        import py2exe
+
+        has_py2exe = True
+    except:
+        has_py2exe = False
+else:
+    has_py2exe = False
+
+#
+# Recipe for ZMQ
+#
+if sys.platform.startswith("win"):
+    #
+    # See http://www.py2exe.org/index.cgi/Py2exeAndzmq
+    # Recipe needed for py2exe to package libzmq.dll
+    os.environ["PATH"] += os.path.pathsep + os.path.split(zmq.__file__)[0]
+
 
 class Install(setuptools.command.install.install):
     def run(self):
+        self.run_command("build_version")
+        self.run_command("build_java_dependencies")
+        setuptools.command.install.install.run(self)
+
+
+class Develop(setuptools.command.develop.develop):
```

Comment:
Yup. I was thinking at one point of moving the code that's in `install` to `build`. The idea would be to stage things like prokaryote and frozen_version in build/lib as part of the build step. That would share between `install` and `py2exe`.